KZL-557 - Add collection controller csharp (#191)
kuzzleio/documentation-V2: src/sdk-reference/csharp/1/collection/update-mapping/index.md

---
layout: sdk.html.hbs
title: updateMapping
description: Update the collection mapping
---

# updateMapping

> Update the collection mapping.

Reviewer: ```suggestion
Updates the collection mapping.
```

> Mapping allow you to exploit the full capabilities of our

Reviewer: ```suggestion
Mapping allows you to exploit the 
```

> persistent data storage layer, [ElasticSearch](https://www.elastic.co/products/elasticsearch) (check here the [mapping capabilities of ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/mapping.html)).

Reviewer: ```suggestion
ElasticSearch [mapping capabilities](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/mapping.html)).
```

## Signature

```csharp
public void updateMapping(string index, string collection, string body);

public void updateMapping(
    string index, 
    string collection, 
    string body, 
    QueryOptions options);

```

## Arguments

| Arguments    | Type    | Description |
|--------------|---------|-------------|
| `index` | <pre>string</pre> | Index name    | 
| `collection` | <pre>string</pre> | Collection name    |
| `mapping` | <pre>string</pre> | JSON string representing the collection data mapping |
| `options` | <pre>Kuzzleio::QueryOptions\*</pre> |  Query options  |

### mapping

A JSON string representing the collection data mapping.

> The mapping must have a root field `properties` that contain the mapping definition:

Reviewer: ```suggestion
The mapping must have a root field `properties` containing the mapping definition:
```
